cpp/src/parquet/encoding-benchmark.cc

```diff
@@ -163,4 +172,132 @@ static void BM_DictDecodingInt64_literals(benchmark::State& state) {
 
 BENCHMARK(BM_DictDecodingInt64_literals)->Range(1024, 65536);
 
+std::shared_ptr<::arrow::Array> MakeRandomStringsWithRepeats(size_t num_unique,
```

Comment:
`MakeRandomStringsWithRepeats` seems like it could be useful elsewhere. Perhaps, this could be moved to somewhere under `arrow/cpp/src/arrow/util/`?

Reply:
I must admit this isn't entirely original and based on a mashup of patterns I saw in other benchmarks and tests.  My thinking is that it would be good to somehow consolidate utilities like this.  Maybe `arrow/testing/random.h` is a good home for utilities like this?  I'll open a new JIRA issue to follow up on this as I think it would have wider downstream impact on the benchmarks and unittests.

Reply:
Created [ARROW-4661](https://issues.apache.org/jira/browse/ARROW-4661) to revisit this.